main/integration-tests/__test__/api-tests/workspace-service-catalogs/create-workspace-service-catalog.test.js

```diff
@@ -166,4 +167,69 @@ describe('Create workspace-service-catalog scenarios', () => {
       });
     });
   });
+  describe('Workspace SC env with studies', () => {
+    it('for EC2Linux should provision correctly', async () => {
+      // const researcher1Session = await setup.createResearcherSession();
+
+      // 1. Create user
+      const admin1Session = await setup.createAdminSession();
+      const username = setup.gen.username();
+      const defaultUser = admin1Session.resources.users.defaults({ username });
+
+      await expect(admin1Session.resources.users.create(defaultUser)).resolves.toMatchObject({
+        username,
+      });
+      // 2. Assign Project ID to user (assigned by default)
+
+      // 3. Create My Study, Org Study
+      const studyIds = [];
+      let studyId = setup.gen.string({ prefix: `create-study-ray-my-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'My Studies' }),
+      ).resolves.toMatchObject({
+        id: studyId,
+      });
+      studyIds.push(studyId);
+
+      studyId = setup.gen.string({ prefix: `create-study-ray-org-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'Organization' }),
+      ).resolves.toMatchObject({
+        id: studyId,
+      });
+      studyIds.push(studyId);
+
+      // 4. Create workspace with the above studies
+      const workspaceName = setup.gen.string({ prefix: 'workspace-sc-test' });
+
+      const env = await admin1Session.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+        envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+        studyIds,
+        description: 'assignment',
+        projectId: setup.defaults.project.id,
+        cidr: '54.240.196.186/32',
+      });
+      expect(env).toMatchObject({
+        name: workspaceName,
+        envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+        envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+        studyIds,
+      });
+
+      // Poll until workspace is provisioned
+      await admin1Session.resources.workflows
+        .versions('wf-provision-environment-sc')
+        .version(1)
+        .findAndPollWorkflow(env.id, 10000, 48);
+
+      // Terminate instance
+      await admin1Session.resources.workspaceServiceCatalogs.workspaceServiceCatalog(env.id).delete();
+      await admin1Session.resources.workflows
```

Comment:
Is there no delay observed between an environment delete API response and the workflow instance actually getting created? Do you think we need a delay added here before we start polling?

Reply:
Ok, is it similar to the deflake delay in collection-resource.js and resource.js? I'll make it 2 seconds as well.

Reply:
Yes, that should be sufficient for now.